File: backend/src/main/kotlin/com/munserv/sectors/domain/SectorSettings.kt
package com.munserv.sectors.domain

import com.munserv.shared.enums.VerificationMode
import com.munserv.shared.types.SectorId
import com.munserv.shared.types.SectorSettingsId
import java.time.Instant

/**
 * Domain entity representing sector-level configuration for Ground Admin and issues.
 *
 * This is a pure domain class with no framework dependencies.
 * All updates return new instances (immutability).
 */
data class SectorSettings(
    val id: SectorSettingsId,
    val sectorId: SectorId,
    val newIssueVerificationMode: VerificationMode,
    val fixVerificationMode: VerificationMode,
    val daysFixedBeforeClosed: Int,
    val minimumGroundAdmins: Int,
    val createdAt: Instant,
    val updatedAt: Instant,
) {
    /**
     * Update newIssueVerificationMode immutably.
     */
    fun withNewIssueVerificationMode(
        mode: VerificationMode,
        now: Instant,
    ): SectorSettings = copy(newIssueVerificationMode = mode, updatedAt = now)

    /**
     * Update fixVerificationMode immutably.
     */
    fun withFixVerificationMode(
        mode: VerificationMode,
        now: Instant,
    ): SectorSettings = copy(fixVerificationMode = mode, updatedAt = now)

    /**
     * Update daysFixedBeforeClosed immutably.
     */
    fun withDaysFixedBeforeClosed(
        days: Int,
        now: Instant,
    ): SectorSettings = copy(daysFixedBeforeClosed = days, updatedAt = now)

    /**
     * Update minimumGroundAdmins immutably.
     */
    fun withMinimumGroundAdmins(
        count: Int,
        now: Instant,
    ): SectorSettings = copy(minimumGroundAdmins = count, updatedAt = now)

    /**
     * Update updatedAt timestamp.
     */
    fun withUpdatedAt(now: Instant): SectorSettings = copy(updatedAt = now)

    companion object {
        private const val DEFAULT_DAYS_FIXED_BEFORE_CLOSED = 7
        private const val DEFAULT_MINIMUM_GROUND_ADMINS = 2

        /**
         * Create default settings for a sector.
         */
        fun createDefault(
            sectorId: SectorId,
            now: Instant,
        ): SectorSettings =
            SectorSettings(
                id = SectorSettingsId.generate(),
                sectorId = sectorId,
                newIssueVerificationMode = VerificationMode.ALL_NOTIFIED,
                fixVerificationMode = VerificationMode.ALL_NOTIFIED,
                daysFixedBeforeClosed = DEFAULT_DAYS_FIXED_BEFORE_CLOSED,
                minimumGroundAdmins = DEFAULT_MINIMUM_GROUND_ADMINS,
                createdAt = now,
                updatedAt = now,
            )
    }
}

File: web/src/shared/types/sectorSettings.ts
/**
 * Verification mode determines how Ground Admins are notified
 */
export type VerificationMode =
  | 'all_notified'
  | 'admin_assigns'
  | 'nearest_auto'
  | 'first_come';

/**
 * Sector settings for Ground Admin verification workflow
 */
export interface SectorSettings {
  id: string;
  sectorId: string;
  newIssueVerificationMode: VerificationMode;
  fixVerificationMode: VerificationMode;
  daysFixedBeforeClosed: number;
  minimumGroundAdmins: number;
  createdAt: string;
  updatedAt: string;
}

/**
 * Request to update sector settings (partial update)
 */
export interface SectorSettingsUpdate {
  newIssueVerificationMode?: VerificationMode;
  fixVerificationMode?: VerificationMode;
  daysFixedBeforeClosed?: number;
  minimumGroundAdmins?: number;
}

/**
 * Map verification modes to their i18n label keys
 */
export const VERIFICATION_MODE_LABELS: Record<VerificationMode, string> = {
  all_notified: 'sectorSettings.modeOptions.allNotified',
  admin_assigns: 'sectorSettings.modeOptions.adminAssigns',
  nearest_auto: 'sectorSettings.modeOptions.nearestAuto',
  first_come: 'sectorSettings.modeOptions.firstCome',
};

/**
 * Verification modes as array for dropdowns
 */
export const VERIFICATION_MODES: VerificationMode[] = [
  'all_notified',
  'admin_assigns',
  'nearest_auto',
  'first_come',
];
